### ISSUE TITLE
Test for JavaScript challenge

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#880]
Please develop a functional test for the #536 . A case for a web site with an index.html referencing non-challengable `<img>` resource must be checked: we must not challenge the image request, the request must be blocked by the module if it doesn't contain the cookie or contain invalid cookie and normally passed if the check was satisfied. Check the timeouts and try to wait bit less than the required delay. Please very carefully follow and test various scenarios from the algorithm described in the original issue. Think about possible protection evasion methods.

The change also affects HTTP sessions handling, so please reimplement `t/unit/test_http_sticky.c` as a functional test and remove the original unit test. Such kind of tests make a lot of headache in support.

**Comment by krizhanovsky** [comment]
Duplicate of #923
